bazel/cc_wrapper.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python
+import os
+import sys
+import tempfile
+
+envoy_real_cc = {ENVOY_REAL_CC}
+envoy_real_cxx = {ENVOY_REAL_CXX}
+
+def sanitize_flagfile(in_path, out_fd):
+    with open(in_path, "rb") as in_fp:
+        for line in in_fp:
+            if line != "-lstdc++\n":
+                os.write(out_fd, line)
+
+def main():
+    compiler = envoy_real_cc
+
+    # Debian's packaging of Clang requires `-no-canonical-prefixes` to print
+    # consistent include paths, but Bazel 0.10 only sets that option at compile
+    # time. We inject it here for the configuration of `@local_config_cc//`.
+    #
+    # https://github.com/bazelbuild/bazel/issues/3977
+    # https://github.com/bazelbuild/bazel/issues/4572
+    # https://bazel-review.googlesource.com/c/bazel/+/39951
+    if sys.argv[1:] == ["-E", "-xc++", "-", "-v"] and "clang" in compiler:
+        os.execv(envoy_real_cxx, [envoy_real_cxx, "-E", "-", "-v", "-no-canonical-prefixes"])
+
+    # `g++` and `gcc -lstdc++` have similar behavior and Bazel treats them as
+    # interchangeable, but `gcc` will ignore the `-static-libstdc++` flag.
+    # This check lets Envoy statically link against libstdc++ to be more
+    # portable between intalled glibc versions.
+    if "-static-libstdc++" in sys.argv[1:]:
+        compiler = envoy_real_cxx
+        argv = []
+        for arg in sys.argv[1:]:
+            if arg == "-lstdc++":
```

Comment:
I'd like to ensure that this is going to work with Envoy OSS-Fuzz that uses `libc++`, seehttps://github.com/google/oss-fuzz/tree/master/projects/envoy. 

Can you checkout OSS-Fuzz (https://github.com/google/oss-fuzz) and run:

```
python infra/helper.py build_image envoy
python infra/helper.py build_fuzzers --sanitizer=address envoy <path to your Envoy source tree>
python infra/helper.py run_fuzzer envoy base64_fuzz_test
```

See https://github.com/google/oss-fuzz/blob/master/docs/new_project_guide.md#testing-locally for the official docs on this flow if you have issues, or reach out to me on Slack.

I think we should eventually put this in a CI slot to ensure we don't regress as we much with build, but for now we need to manually do this.

Reply:
Running now. I've only got a laptop to build with so it's taking a pretty good amount of time.

Reply:
oss-fuzz works with the new script, but we need to add `python` to the Debian package lits in `oss-fuzz/projects/envoy/Dockerfile`.

Reply:
Correction: I ran the oss-fuzz script _correctly_ this time (on my branch instead of envoy upstream) and it failed with some C++ stdlib linker problem. I think a `-lstdc++` was getting through the wrapper somehow. Rebuilding now to see if more filtering fixed it.

Reply:
Verified it works once we add `-stdlib=libc++` to the list of flags that need `-lstdc++` to be pruned first.